### PR TITLE
feat: add version skew detection POC

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

POC for version skew detection between Central and Sensor. Adds a `VersionSkew` field to `ClusterStatus` that reports whether the version skew is matching, compatible, or incompatible. Central computes this on sensor connect using a hardcoded 3-minor-version range and persists it. Data flows through `/v1/clusters` API — visible in browser console, no UI component changes.

## User-facing documentation

- [ ] [CHANGELOG.md](https://github.com/stackrox/stackrox/blob/master/CHANGELOG.md) is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

## Testing and quality

- [ ] the change is production ready: the change is [GA](https://github.com/stackrox/stackrox/blob/master/PR_GA.md), or otherwise the functionality is gated by a [feature flag](https://github.com/stackrox/stackrox/blob/master/pkg/features/README.md)
- [ ] CI results are [inspected](https://docs.google.com/document/d/1d5ga073jkv4CO1kAJqp8MPGpC6E1bwyrCGZ7S5wKg3w/edit?tab=t.0#heading=h.w4ercgtcg0xp)

### Automated testing

- [x] added unit tests

### How I validated my change

Unit tests for `ComputeVersionSkew` covering: matching versions, compatible skew (1-2 minor behind), incompatible skew (3+ minor behind, sensor ahead, cross-major), edge cases (empty versions, dev builds, low minor clamping). Compilation verified for affected packages.

### CODE REVIEW

File: ui/apps/platform/src/types/cluster.proto.ts [hunk 1]
```diff
@@ -157,13 +157,32 @@ export type ClusterCertExpiryStatus = {
...
+export type VersionSkewStatus =
+    | 'VERSION_SKEW_STATUS_UNSPECIFIED'
+    | 'VERSION_SKEW_STATUS_MATCHING'
+    | 'VERSION_SKEW_STATUS_COMPATIBLE'
+    | 'VERSION_SKEW_STATUS_INCOMPATIBLE';
+
+export type VersionSkewReason =
+    | 'VERSION_SKEW_REASON_UNSPECIFIED'
+    | 'VERSION_SKEW_REASON_SENSOR_TOO_OLD'
+    | 'VERSION_SKEW_REASON_SENSOR_AHEAD';
```
Comment (pedrottimark) [inline ui/apps/platform/src/types/cluster.proto.ts]: Parallel wording for `TOO_OLD` versus `AHEAD` seems worthwhile.

Comment (pedrottimark) [reply ui/apps/platform/src/types/cluster.proto.ts]: Incompatible per MIsha

File: ui/apps/platform/src/types/cluster.proto.ts [hunk 2]
```diff
@@ -157,13 +157,32 @@ export type ClusterCertExpiryStatus = {
...
+
+export type VersionSkew = {
+    status: VersionSkewStatus;
+    reason: VersionSkewReason;
+    minCompatibleSensorVersion: string;
+    maxCompatibleSensorVersion: string;
```
Comment (pedrottimark) [inline ui/apps/platform/src/types/cluster.proto.ts:177]: x.y but not z

File: proto/storage/cluster.proto [hunk 3]
```diff
@@ -223,6 +223,29 @@ message ClusterCertExpiryStatus {
...
+  VERSION_SKEW_INCOMPATIBILITY_REASON_UNSPECIFIED = 0;
+  VERSION_SKEW_INCOMPATIBILITY_REASON_SENSOR_TOO_OLD = 1;
+  VERSION_SKEW_INCOMPATIBILITY_REASON_SENSOR_TOO_NEW = 2;
+}
+
+message VersionSkew {
+  VersionSkewStatus status = 1;
+  // Only set when status is VERSION_SKEW_STATUS_INCOMPATIBLE.
+  VersionSkewIncompatibilityReason incompatibility_reason = 2;
+  // Oldest compatible Sensor version, in X.Y format (patch version not included).
```
Comment (pedrottimark) [inline proto/storage/cluster.proto:243]: Remind me what you think about pro and con of Sensor version in both `clusters` and `metadata` endpoints?

If you keep it in `metadata` proto, can you add the comments there?

As I said, metadata (only) works for UI, if better from viewpoint of central.
* **System Health** page depends only on `status` property
* **Clusters** page depends on central `version` from `metadata` endpoint which UI polls and data available in any UI component that needs it and `sensorVersion` from `clusters` endpoint that the page polls.

For your info, pending decision where `VersionSkew` object fits, `Cluster` has a `status` property, whose `upgradeStatus` property will become obsolete at some time in the future, and will be superseded in this effort by sensor compatibility status.
https://github.com/stackrox/stackrox/blob/master/proto/storage/cluster.proto#L192

Name is not a problem from my viewpoint, because will access as `whatever.status`

Consider this contribution approved from UI viewpoint assuming you and MIsha will drive to conclustion.
